core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -425,6 +426,8 @@ public ManagedChannelImpl shutdown() {
       maybeTerminateChannel();
       if (!terminated) {
         transportsCopy.addAll(transports.values());
+        transports.clear();
+        decommissionedTransports.addAll(transportsCopy);
```

Comment:
Please delete "due to channel idleness" from the javadoc of decommissionedTransports.